### PR TITLE
by yanniboi: Added submission entity to replace submission node type.

### PR DESCRIPTION
**Pull request by yanniboi**

### CODE REVIEW

**Comment by patrickd-** [comment]
looked over the code, certainly looks better than anything i could have come up with in that amount of time 😉
d8 feels like a lot more boilerplate stuff though, a lot more complex than anything you'd see in eg. a rails project

only thing that caught my eye is that i don't quite get why that debug message is here 
(https://github.com/yanniboi/simplytest/blob/1704ebf97acabae904ee303eb02d3c9c3158ff9a/modules/simplytest_submission/src/SubmissionService.php#L126) but yeah.. ¯\\\_(ツ)\_/¯

dunno if you already come up with something like that but I think we should abstract the whole script stuff into something like this 
```
class ShellScript
 constructor() adds '#!/bin/bash ...'
 appendScript(ShellScript) for appending another ShellScript to this one
 installPackage(String) appends 'apt install String [..]'
 writeFile(String, Array) writes a file echo>> with multiple lines of content
 overWriteFile(String, Array) writes a file echo> with multiple lines of content
 execute(String, Array) execute command with properly sanitized parameters
```

File: modules/simplytest_submission/src/SubmissionService.php [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+   *   The entity that the script is being build for.
+   *
+   * @return bool|array
+   *   Deploy script for building submission instance.
+   */
+  public function buildScript(EntityInterface $entity) {
+    $this->buildEntity = $entity;
+    $this->debug('No results from spawn.sh.');
+    $this->debug($entity, 'Entity object.');
+
```
Comment (yanniboi) [inline modules/simplytest_submission/src/SubmissionService.php]: Good catch!! I made a note of the format of the debug method to remind myself and forgot to remove it :P 

Will remove.

**Comment by yanniboi** [comment]
With regard to your "class ShellScript...", yes this sounds like a great idea, but lets do that as a follow up and get this committed first.

**Comment by patrickd-** [comment]
how can i launch opensocial?
`/submission/add/?project=social`
didn't work, or i'm stupid right now :)

besides that we're good to go!

non-blockers:
* We should turn it into a profile to have it ready after install (set home page to /submission/add, default sponsors block with logos, ...?)
* I'm able to try launching sandboxes although spawn.sh token hasn't been set (progress page is shown with websocket errors in console, watchdog logs: No settings found for simplytest_submission)
* submission/x is empty and should probably be submission/x/status?
* install project option is disabled, does this no longer work here?
* title of the submission page is ... "Thank you*r* for your submission" :D maybe change it to something like submission is being processed?
* submit button is now "deploy sandbox" – i personally don't disagree but i'm not sure if this would be understood by non-devs?

**Comment by yanniboi** [comment]
I've fixed the query parameter regression and logged the non-blockers for followups here:
https://github.com/yanniboi/simplytest/issues

**Comment by yanniboi** [comment]
Also created an issue on d.o for this (https://www.drupal.org/node/2878179), however that is just for reference if someone ever comes looking through the issues. I will also create a single squashed commit for the 8.x-3.x branch when this is ready so we don't get a lot of small tweaks in the git history.

**Pull request merged**